coalib/output/ConsoleInteraction.py

```diff
@@ -199,6 +199,112 @@ def acquire_actions_and_apply(console_printer,
             break
 
 
+def get_context(console_printer, file_dict,
+                sourcerange):  # pragma: no cover
+    """
+    Gets context for the affected lines and also the line numbers
+    of the context
+
+    :param file_dict:       A dictionary containing all files as values with
+                            filenames as key.
+    :param sourcerange:     The SourceRange object referring to the related
+                            lines to print.
+
+    :yield:                 Affected file's contextual information
+
+    """
+
+    related_lines = list()
+    related_lines_nr = list()
+
+    affected_line = file_dict[sourcerange.file][
+                    sourcerange.start.line-1].rstrip()
+    affected_line = affected_line.replace('\t', 8*' ')
+
+    indet_space_count = len(affected_line)-len(affected_line.lstrip())
+    if indet_space_count > 0:
+        for j in range(sourcerange.start.line-2, -1, -1):  # pragma: no cover
+            rline = file_dict[sourcerange.file][j].rstrip()
+            affected_line = rline.replace('\t', 8*' ')
+            prv_space_count = len(affected_line) - len(affected_line.lstrip())
+            if prv_space_count < indet_space_count and rline:
+                related_lines.append(rline)
+                related_lines_nr.append(j+1)
+                indet_space_count = prv_space_count
+            if indet_space_count == 0:
+                break
+    else:
+        pre_context = list(range(sourcerange.start.line-1))
+        pre_context.reverse()
+        pre_context = pre_context[0: min(2, len(pre_context))]
+        for i in pre_context:
+            related_lines.append(file_dict[sourcerange.file][i])
+            related_lines_nr.append(i+1)
+
+    related_lines_nr.reverse()
+    related_lines.reverse()
+
+    return related_lines, related_lines_nr
+
+
+def print_before_context(console_printer,
+                         file_dict,
+                         sourcerange):  # pragma: no cover
+    """
+    Prints the context for the affected lines.
+
+    :param console_printer: Object to print messages on the console.
+    :param file_dict:       A dictionary containing all files as values with
+                            filenames as key.
+    :param sourcerange:     The SourceRange object referring to the related
+                            lines to print.
+    """
+
+    related_lines, related_line_nr = get_context(console_printer, file_dict,
+                                                 sourcerange)
+    no_color = not console_printer.print_colored
+    j = 0
```

Comment:
Please change name of variable `j` so it becomes easier to understand what it is doing